File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/domain/workout/ExerciseDetails.kt
package pt.isel.leic.ptgest.domain.workout

import pt.isel.leic.ptgest.domain.common.ExerciseType
import pt.isel.leic.ptgest.domain.common.MuscleGroup

data class ExerciseDetails(
    val id: Int,
    val name: String,
    val description: String?,
    val muscleGroup: MuscleGroup,
    val type: ExerciseType,
    val ref: String?
)

File: code/js/src/views/user/TrainerViews/models/exercises/ExerciseDetails.ts
class ExerciseDetails {
    id: number
    name: string
    description: string | null
    muscleGroup: string[]
    modality: string
    ref: string
    constructor(id: number, name: string, description: string | null, muscleGroup: string[], modality: string, ref: string) {
        this.id = id
        this.name = name
        this.description = description
        this.muscleGroup = muscleGroup
        this.modality = modality
        this.ref = ref
    }
}

export default ExerciseDetails
